connectors/src/connectors/microsoft/lib/graph_api.ts

```diff
@@ -0,0 +1,84 @@
+import type { Client } from "@microsoft/microsoft-graph-client";
+import type * as MicrosoftGraph from "@microsoft/microsoft-graph-types";
+
+export async function getSites(client: Client) {
+  const res = await client
+    .api("/sites?search=*")
+    .select("id,name,displayName")
+    .get();
+  return res.value as MicrosoftGraph.Site[];
+}
+
+export async function getDrives(client: Client, siteId: string) {
+  const res = await client
+    .api(`/sites/${siteId}/drives`)
+    .select("id,name")
+    .get();
+  return res.value as MicrosoftGraph.Drive[];
+}
+
+export async function getFilesAndFolders(
+  client: Client,
+  driveId: string,
+  parentId?: string
+) {
+  const parent = parentId ? `items/${parentId}` : "root";
+  const res = await client
+    .api(`/drives/${driveId}/${parent}/children`)
+    .select("id,displayName")
+    .get();
+  return res.value as MicrosoftGraph.DriveItem[];
```

Comment:
I see we have "as" everywhere below
Maybe they have io-ts like utils to parse api results as typed params?